include/aws/common/log.h

```diff
@@ -0,0 +1,120 @@
+#ifndef AWS_COMMON_LOG_H
+#define AWS_COMMON_LOG_H
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <stdarg.h>
+
+#include <aws/common/memory_pool.h>
+
+enum aws_log_level
+{
+    AWS_LOG_LEVEL_OFF = 0, /* No logs will be processed at all. */
+    AWS_LOG_LEVEL_FATAL,
+    AWS_LOG_LEVEL_ERROR,
+    AWS_LOG_LEVEL_WARN,
+    AWS_LOG_LEVEL_INFO,
+    AWS_LOG_LEVEL_DEBUG,
+    AWS_LOG_LEVEL_TRACE
+};
+
+#ifndef AWS_LOG_LEVEL
+#define AWS_LOG_LEVEL AWS_LOG_LEVEL_TRACE
+#endif /* AWS_LOG_LEVEL */
+
+#ifdef _WIN32
+#pragma warning(disable:4127) /* Suppress warnings for const conditionals (for the below LOG macros). */

```

Comment:
if this absolutely needs to be in a header file, you prolly need to do pragma push and pops

Also, this is msvc specific, not windows specific. If you used mingw on windows, this would be irrelevant (and broken)